articles/greedy-algorithms/index.md

```diff
@@ -0,0 +1,161 @@
+# Optimizing Stock Price Profit Using Greedy Algorithms
+
+Stock markets are where buyers and sellers represent ownership for the businesses listed on the markets. Many have become millionaires by trading. In today's era, many companies offer solutions to test out our strategies on previous data. This is called back-testing. If the strategies return good profits during back-testing, it can be deployed in the live markets.
+
+One of the aspects of back-testing is to know the maximum possible profit. We compute the maximum potential gain from stock prices and compare the strategy's performance to the maximum profit possible. Once the back-testing method is at par with the maximum profit, it is deployed into the real-time analysis.
+
+### Problem Statement
+
+Given a list of prices over a timeframe, we compute the maximum possible profit possible through performing one buy operation or one sell operation at a given time. At a given timestamp, only one of the two operations can be used. The total number of buying and selling operations that can be performed are unlimited. 
+
+### Greedy Approach 
+
+We will be using the greedy method to obtain the maximum possible profit. The [greedy method](https://en.wikipedia.org/wiki/Greedy_algorithm) is a type of problem-solving strategy, where the best possible solution at the current instance is chosen. Unlike other algorithms, which consider the optimal solution over a more extensive timeframe, greedy algorithms make decisions at the given time instance. This results in efficient computations and faster results. However, there is a downside to this approach. The greedy approach might not always provide the most optimal solution. This occurs because it considers only the current instance to make a decision. 
+
+In this problem, we will understand the nature of greedy algorithms and determine the maximum possible profit. Let's begin.
+
+### An introduction to the Greedy Solution
+
+In this problem, the greedy method is used to maximize the profit given a list of values. The greedy algorithm obtains the profit by going with the amount that looks best at each timestamp. This project's scope is limited to previous data and can not model the fluctuations and variations in stock markets. We assume that pre-historic data is available and are thus proceeding with the same.
+
+### Given Input
+
+The input to the program is a list of historical values (integers). It is stored in an array. Python's list is an efficient implementation of the dynamic array. The advantage of the dynamic array is that it grows automatically when additional elements are inserted, and no space is available for the new element. The amortized cost of insertion of an element into a dynamic array is O(1), whereas the worst case is still Θ(n).
```

Comment:
> O(1), whereas the worst case is still Θ(n)

It would be better if you could give a one-liner about Big Oh vs Omega so that readers who are beginners aren't overwhelmed.